src/databricks/labs/ucx/workspace_access/base.py

```diff
@@ -32,3 +35,12 @@ def get_apply_task(self, item: Permissions, migration_state: MigrationState) ->
     @abstractmethod
     def object_types(self) -> set[str]:
         """This method returns a set of strings, that represent object types that are applicable by this instance."""
+
+    @abstractmethod
+    def verify(
+        self,
+        object_type: str,
+        object_id: str,
+        acl: list[iam.AccessControlRequest | sql.AccessControl | iam.ComplexValue] | Grant | workspace.AclItem,
+    ) -> bool:
```

Comment:
```suggestion
    def verify(self, item: Permissions) -> bool:
```
You can't add union type with child-related classes here, because it leaks implementation details upwards and common sense doesn't allow us to merge this as it is now. It simply means no sense and breaks the dependency flow, increasing coupling.

Please rewrite the underlying implementations, the way both applier tasks and verify call a common private method. And implementation of verify() converts Permission into a call of that private method.

I also don't see any external calls of this verify() method - why do we need it, again?

Reply:
the external call could be used if we want to build cli command for verification